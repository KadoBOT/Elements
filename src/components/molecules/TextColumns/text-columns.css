@media screen and (max-width: 480px) {
  .text-columns {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
  }

  .text-columns > p {
    flex: 1;
    font-size: 0.7em;
    text-align: center !important;
    margin: 0px 8vw;
    border-left: none !important;
    border-bottom: 1px solid rgba(128, 128, 128, 0.23);
    padding: 3vw 0px 3vw 0px !important;
  }
}

.text-columns {
  display: flex;
  margin: 3vw 0px 5vw 0px;
}

.text-columns > p {
  padding: 0px 3vw;
  text-align: left;
  flex: 1;
}

.text-columns > p:not(:first-child) {
  border-left: 1px solid rgba(214, 83, 112, 0.13);
}
