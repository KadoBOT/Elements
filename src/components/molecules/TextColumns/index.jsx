import React, {PropTypes} from 'react'
import R from 'ramda'

import { Description } from '../../atoms/Text'
import './text-columns.css'

const TextColumns = ({city}) => {
  return(
    <div className="text-columns">
      {R.map(t =>
        <Description>{t}</Description>
        , city.texts)}
    </div>
  )
}

TextColumns.propTypes = {
  props: PropTypes.type
}

export default TextColumns
