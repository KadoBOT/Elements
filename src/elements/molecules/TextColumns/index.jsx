import React, {PropTypes} from 'react'
import R from 'ramda'

import { Description } from '../../atoms/Text'
import './text-columns.css'

const TextColumns = ({city}) => (
  <div className="text-columns">
    {R.map(t =>
      <Description key={t}>{t}</Description>
      , city.texts)}
  </div>
)

TextColumns.propTypes = {
  city: PropTypes.shape({
    texts: PropTypes.array
  }).isRequired
}

export default TextColumns
