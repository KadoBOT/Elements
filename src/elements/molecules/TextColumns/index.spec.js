import React from 'react'
import { shallow, render, mount } from 'enzyme';

import TextColumns from './'

it('should render the TextColumns', () => {
    const wrapper = shallow(
        <TextColumns city={{texts: []}} />
    );
    expect(wrapper).toMatchSnapshot();
});
