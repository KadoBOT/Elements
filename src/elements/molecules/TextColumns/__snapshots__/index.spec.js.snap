exports[`test should render the TextColumns 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="text-columns">
    
</div>,
  "nodes": Array [
    <div
      className="text-columns">
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        city={
                Object {
                        "texts": Array [],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "city": Object {
            "texts": Array [],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="text-columns">
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="text-columns">
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    city={
        Object {
            "texts": Array [],
          }
    } />,
}
`;
